ngrinder-controller/src/main/java/org/ngrinder/perftest/service/PerfTestService.java

```diff
@@ -628,14 +634,30 @@ public GrinderProperties getGrinderProperties(PerfTest perfTest, ScriptHandler s
 	 * @param perfTest perfTest
 	 * @return File location in which the perftest script and resources are distributed.
 	 */
-	public ScriptHandler prepareDistribution(PerfTest perfTest) {
+	public ScriptHandler prepareDistribution(PerfTest perfTest) throws IOException {
 		File perfTestDistDirectory = getDistributionPath(perfTest);
 		User user = perfTest.getCreatedUser();
-		FileEntry scriptEntry = checkNotNull(
+		String scm = perfTest.getScm();
+		FileEntry scriptEntry;
+
+		if (isGitHubScript(scm)) {
+			String gitHubConfigName = extractGitHubConfigName(scm);
+			String scriptName = perfTest.getScriptName();
+
+			GitHubConfig gitHubConfig = gitHubFileEntryService.getGitHubConfig(user, gitHubConfigName);
+			GitHub gitHub = gitHubFileEntryService.getGitHubClient(gitHubConfig);
+			GHRepository ghRepository = gitHub.getRepository(gitHubConfig.getOwner() + "/" + gitHubConfig.getRepo());
+
+			gitHubFileEntryService.checkoutGitHubScript(user, ghRepository, gitHubConfig, scriptName);
+			scriptEntry = gitHubFileEntryService.getOne(user, ghRepository, gitHubConfigName, scriptName);
```

Comment:
I dont' think that PerftestService have to know about either `GithubClient` or `GHRepository`. Is it possible to change `githubFileEntryService.getOne(user, githubConfig, scriptName)`?

Reply:
To get `GHRepository` it needs to github api call.
so. i reuse this to reduce api call.